uvicorn/middleware/wsgi.py

```diff
@@ -46,43 +59,52 @@ def build_environ(scope, message, body):
             corrected_name = "HTTP_%s" % name.upper().replace("-", "_")
         # HTTPbis say only ASCII chars are allowed in headers, but we latin1
         # just in case
-        value = value.decode("latin1")
+        value: str = value_.decode("latin1")
         if corrected_name in environ:
             value = environ[corrected_name] + "," + value
         environ[corrected_name] = value
     return environ
 
 
 class WSGIMiddleware:
-    def __init__(self, app, workers=10):
+    def __init__(self, app: ASGI3Application, workers: int = 10):
         self.app = app
         self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=workers)
 
-    async def __call__(self, scope, receive, send):
+    async def __call__(
+        self, scope: HTTPScope, receive: ASGIReceiveCallable, send: ASGISendCallable
```

Comment:
What's the error you have below? 
```suggestion
        self, scope: Scope, receive: ASGIReceiveCallable, send: ASGISendCallable
```

Reply:
There isn't any error in this line.
Also, I think `HTTPScope` should be fine as it mentions this one line down:
https://github.com/encode/uvicorn/blob/a8af0da26c921b27849701f2cedbf89b8f3cbd86/uvicorn/middleware/wsgi.py#L62

Reply:
Yeah, but the line below says that it should be `HTTPScope` **after** it, not before. Otherwise, we'd not need to assert it.

Reply:
The `assert` has been put just after the function header - in the first line.
https://github.com/encode/uvicorn/blob/a8af0da26c921b27849701f2cedbf89b8f3cbd86/uvicorn/middleware/wsgi.py#L61-L62
My reasoning was that this should mean that the function is expecting the type of scope as HTTP, and therefore I put the typehint for that.

Also, I tried to replace that with `Scope`. That threw new mypy errors as the rest of the code is expecting attributes on `scope` which are only there in `HTTPScope` and not in `Scope`.